core/src/main/scala/org/apache/spark/deploy/RRunner.scala

```
@@ -25,22 +25,23 @@ import scala.collection.JavaConverters._
 import org.apache.hadoop.fs.Path
 
 import org.apache.spark.{SparkException, SparkUserAppException}
+import org.apache.spark.api.conda.CondaEnvironment
 import org.apache.spark.api.r.{RBackend, RUtils, SparkRDefaults}
 import org.apache.spark.util.RedirectThread
 
 /**
  * Main class used to launch SparkR applications using spark-submit. It executes R as a
  * subprocess and then has it connect back to the JVM to access system properties etc.
  */
-object RRunner {
-  def main(args: Array[String]): Unit = {
+object RRunner extends CondaRunner {
+  override def run(args: Array[String], maybeConda: Option[CondaEnvironment]): Unit = {
     val rFile = PythonRunner.formatPath(args(0))
 
     val otherArgs = args.slice(1, args.length)
 
     // Time to wait for SparkR backend to initialize in seconds
     val backendTimeout = sys.env.getOrElse("SPARKR_BACKEND_TIMEOUT", "120").toInt
-    val rCommand = {
+    val rCommand = maybeConda.map(_.condaEnvDir + "/bin/r").getOrElse {
```

Unless we want to test and support this, maybe we should just throw an exception if maybeConda is not None.